libretroshare/src/serialiser/rstypeserializer.h

```diff
@@ -290,11 +290,11 @@ struct RsTypeSerializer
 				// Use same allocator to avoid deep copy
 				RsGenericSerializer::SerializeContext kCtx(
 				            nullptr, 0, ctx.mFlags, &allocator );
-				serial_process<T>(j, kCtx, const_cast<T&>(kv.first), "key");
```

Comment:
originally I think that the <T> was here to ensure that the correct template woul be used. There are situations where the type can be implicitely converted (e.g. time_t -> unsigned int) in different ways depending on the system where it happens, which does not guarranty that the serialization will be correct. So I would keep the template just in case.

Reply:
Those cases should not happen if a template match the type, as an example even for inheriting types the compiler generate a specific function for each type and doesn't apply the function of super-type. @sehraf did you removed the <T> part because was breaking nested STL containers? If it is the case the <T> should be removed anyway.

Reply:
Why is sometimes serial_process\<X\>(...) used and sometimes not? std::set is using it and std::list not? JSON is mostly not using it at all. That's above my head ...

Without this change i got this error during linkage:
```
/usr/bin/ld: temp/linux-g++/obj/MessagesDialog.o: in function `void RsTypeSerializer::serial_process<unsigned int, std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int> >(RsGenericSerializer::SerializeJob, RsGenericSerializer::SerializeContext&, std::map<unsigned int, std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int>, std::less<unsigned int>, std::allocator<std::pair<unsigned int const, std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int> > > >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/user/Projects/RetroShare-build/retroshare-gui/src/../../../RetroShare/libretroshare/src/serialiser/rstypeserializer.h:156: undefined reference to `bool RsTypeSerializer::to_JSON<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int> >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int> const&, rapidjson::GenericDocument<rapidjson::UTF8<char>, rapidjson::MemoryPoolAllocator<rapidjson::CrtAllocator>, rapidj
```
The Problem is that it's trying to use *bool RsTypeSerializer::**to_JSON**<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, unsigned int> >* while it sould be  **serial_process**<std::pair<....